Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -129,9 +130,9 @@ private void SetupAxiomBuilder(VCExpressionGenerator gen)
       }
     }
 
-    void SetupProcess()
+    void SetupProcess(bool recover = false)
```

Comment:
It seems like you don't need the additional `recover` argument if you run `Process = null;` before calling `SetupProcess` on line 660.

Reply:
I followed here the principle Don't Repeat Yourself. Looking at this pattern used twice, It thought it would be better to refactor it:
`Process = null; SetupProcess();` ==> `SetupProcess(/*recover*/true)`
I see several reasons that this enhances the readability and maintainability
- Having a statement controlling the inner if of another hidden statement is error-prone and hard to follow.
- If Process is implemented in a different way down the road, we will have to change all statements before call sites
- Process is assigned twice in the first statement (once null, once the new process), but what we want here is just discard the previous process and only one assignment can be made.

Because of the purpose of this PR is not maintainability, if you tell me that you prefer the first version, I'd happily make the change if this gets this PR accepted more easily.

Reply:
I suggest inlining the

```
if (Process != null && !recover)
      {
        return;
      }
```

Part of the method, since only one call passes `recover == false`.

Reply:
That's a good point. Since there is never a point where we call `SetupProcess` "just in case", I removed all ifs and just assume that if the process was there, it should be restarted anyway with this call.